src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs

```diff
@@ -561,6 +563,25 @@ private async Task<ResourceConfig<VirtualMachineScaleSet>> SimpleParameterSetOrc
 
                 var hostGroup = resourceGroup.CreateDedicatedHostGroupSubResourceFunc(_cmdlet.HostGroupId);
 
+                Dictionary<string, List<string>> auxAuthHeader = null;
+                if (!string.IsNullOrEmpty(_cmdlet.ImageReferenceId))
```

Comment:
should you only fill the auxheader when ImageReferenceId is passed? Should the auxheader be filled from the user in any other case?

Reply:
I remember now. Currently users cannot pass in aux header in PS.  We are auto populating aux header with correct token when user references a gallery image that is cross tenant. so currently in powershell aux header is only used in this specific case